src/Lender.sol

```diff
@@ -546,45 +598,43 @@ contract Lender {
-        uint256 returned = convertDecimals(u, pt, obtained);
+        uint256 returned = convertDecimals(u, _Market.tokens[p], obtained);
 
         // Mint Illuminate PTs to msg.sender
         IERC5095(principalToken(u, m)).authMint(msg.sender, returned);
         emit Lend(p, u, m, returned, spent, msg.sender);
-        return returned;
+        return (returned);
     }
 
-    // @notice: Handles all lending to ETH markets and any swaps necessary
-    // @param adapter: The adapter contract to lend through
+    // @notice: Adjusts all Swivel Amounts according to the slippageRatio
+    // @param slippageRatio: The slippageRatio to adjust by
     // @param d: The calldata for the lend (and potentially swap) operation
-    function ETHLend(address adapter, bytes calldata d) internal returns (bool success, bytes memory returndata) {
-        // Parse the calldata
-        (
-            address underlying,
-            uint256 maturity,
-            address pool,
-            uint256 amount,
-            uint256 minimum,
-            address lst,
-            uint256 swapMinimum
-        ) = abi.decode(d, (address, uint256, address, uint256, uint256, address, uint256));
+    // @returns The adjusted amount array
+    function adjustSwivelAmounts(uint256[] memory amounts, uint256 slippageRatio) internal pure returns (uint256[] memory) {
+        for (uint256 i; i != amounts.length; ) {
+            amounts[i] = amounts[i] - amounts[i] * slippageRatio;
+            unchecked {
+                ++i;
+            }
+        }
+        return (amounts);
+    }
+
+    // @notice: Handles all lending to ETH markets and any swaps necessary
+    // @param lst: The address of the token to swap to
+    // @param amount: The amount of underlying to spend
+    // @param swapMinimum: The minimum amount of lst to receive
+    // @returns lent: The amount of underlying to be lent
+    // @returns slippageRatio: The slippageRatio of the swap (1e18 based % to adjust swivel orders if necessary)
+    function ETHWrap(address lst, uint256 amount, uint256 swapMinimum) internal returns (uint256 lent, uint256 slippageRatio) {
         // If the "lst" address is populated, a swap is required
         if (lst != address(0)) {
-            amount = ICurveWrapper(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2).swap(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2, lst, amount, swapMinimum);
-            // Encode to the adapter interface
-            bytes memory d_ = abi.encode(underlying, maturity, pool, amount, minimum);
-            // Conduct the lend operation to acquire principal tokens
-            (success, returndata) = adapter.delegatecall(
-                abi.encodeWithSignature('lend(bytes calldata inputdata)', d_));
-            return (success, returndata);
+            (lent, slippageRatio)  = ICurveWrapper(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2).swap(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2, lst, amount, swapMinimum);
```

Comment:
move `0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2` to constants

Reply:
yeah, i agree having a constant called `WETH` would be a lot easier to read...